src/NServiceBus.Core/Pipeline/Incoming/MessageTypeEnricherBehavior.cs

```diff
@@ -0,0 +1,19 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using Pipeline;
+
+    class MessageTypeEnricherBehavior : Behavior<IIncomingLogicalMessageContext>
```

Comment:
Should the name (of the behavior and the associated feature) be named more explicitly? There is already an existing behavior that is responsible for setting the message type header on outgoing messages. Should this one be named something like `InferredMessageTypeEnricher` or something?

Reply:
Done